app/lib/audit/backend.dart

```diff
@@ -73,3 +94,9 @@ class AuditBackend {
     );
   }
 }
+
+extension RequestHeaderBeforeExt on shelf.Request {
+  DateTime? get before => requestedUri.queryParameters.containsKey('before')
+      ? DateTime.tryParse(requestedUri.queryParameters['before']!)
```

Comment:
Let's throw 400 if there is a parse error here..

Reply:
I've updated the code to return a formatted 404. Right now we don't do such parsing (yet) in other places, e.g. https://pub.dev/packages?q=json&page=2x gets a valid response :(
I think it may not be important to spell out which parameter is wrong, so reusing the 404 may be fine?